libs/http-service/src/http-client.cpp

```diff
@@ -64,8 +67,17 @@ Request::Ptr HttpClient::request(const Request::Ptr& request)
         json::object({{"requests", json::array({request->toJson()})}}).dump(),
         "application/json");
 
-    if (tileResponse && tileResponse->status == 200)
-        reader->read(tileResponse->body);
+    if (tileResponse) {
+        if (tileResponse->status == 200) {
+            reader->read(tileResponse->body);
+            request->setStatus(mapget::RequestStatus::Success);
```

Comment:
This will lead to onDone_ being invoked twice, because the tileResponse already sets its status to Success as soon as it has received the expected number of tiles.

Reply:
Ah I see, because of the `notifyResult` callback above 👍